app/assets/javascripts/components/overview/statistics_update_info.jsx

```diff
@@ -1,27 +1,124 @@
-    return <div />;
-  }
-  const lastUpdate = course.updates.last_update.end_time;
-  const lastUpdateMoment = moment.utc(lastUpdate);
-  const averageDelay = course.updates.average_delay;
-  let lastUpdateMessage;
-  if (lastUpdate) {
-    lastUpdateMessage = `${I18n.t('metrics.last_update')}: ${lastUpdateMoment.fromNow()}`;
-  }
-  const nextUpdateExpectedTime = lastUpdateMoment.add(averageDelay, 'seconds');
-  let nextUpdateMessage;
-  if (nextUpdateExpectedTime.isAfter()) {
-    nextUpdateMessage = `. ${I18n.t('metrics.next_update')}: ${nextUpdateExpectedTime.fromNow()}`;
+const StatisticsUpdateInfo = createReactClass({
+  displayName: 'StatisticsUpdateInfo',
+
+  propTypes: {
+    course: PropTypes.object.isRequired,
+  },
+
+  getInitialState() {
+    return {
+      showModal: false
+    };
+  },
+
+  getUpdateTimesInformation() {
+    const course = this.props.course;
+    const lastUpdate = course.updates.last_update.end_time;
+    const lastUpdateMoment = moment.utc(lastUpdate);
+    const averageDelay = course.updates.average_delay;
+    let lastUpdateMessage = '';
+    let nextUpdateMessage = '';
+    let isNextUpdateAfter = null;
+
+    if (lastUpdate) {
+      lastUpdateMessage = `${I18n.t('metrics.last_update')}: ${lastUpdateMoment.fromNow()}`;
+      const nextUpdateExpectedTime = lastUpdateMoment.add(averageDelay, 'seconds');
+      isNextUpdateAfter = nextUpdateExpectedTime.isAfter();
+      nextUpdateMessage = `${I18n.t('metrics.next_update')}: ${nextUpdateExpectedTime.fromNow()}`;
+    }
+
+    return [lastUpdateMessage, nextUpdateMessage, isNextUpdateAfter];
+  },
+
+  // Update numbers (ids) are stored incrementally as counts in update_logs, so the
+  // last update number is the total number of updates till now
+  getTotalNumberOfUpdates() {
+    const updateNumbers = Object.keys(this.props.course.flags.update_logs);
+
+    return updateNumbers[updateNumbers.length - 1];
+  },
+
+  getCourseUpdateErrorMessage() {
+    const errorCount = this.props.course.updates.last_update.error_count;
+
+    return `${I18n.t('metrics.error_count_message', { error_count: errorCount })} `;
+  },
+
+  toggleModal() {
+    this.setState({
+      showModal: !this.state.showModal
+    });
+  },
+
+  render() {
+    const course = this.props.course;
+
+    if ((Features.wikiEd && !course.ended) || !course.updates.last_update) {
+      return <div />;
+    }
+
+    const [lastUpdateMessage, nextUpdateMessage, isNextUpdateAfter] = this.getUpdateTimesInformation();
+    const updateTimesMessage = isNextUpdateAfter ? `${lastUpdateMessage}. ${nextUpdateMessage}. ` : `${lastUpdateMessage}. `;
+
+    // If no errors, display only update time information
+    if (course.updates.last_update.error_count === 0) {
+      return (
+        <div className="pull-right mb2">
+          <small>{ updateTimesMessage }</small>
+        </div>
+      );
+    }
+
+    // If there are errors
+
+    // Render Modal
```

Comment:
There's too much going on in this component. The modal should be extracted as a separate component.

Reply:
Done!